### PR TITLE
[quantization] Support multi-turn workflows

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit supports multi-turn workflows.

```bash
python -m tico.quantization.recipes.debug.static_llama_runtime \
  --model /mnt/sdb/model/Llama-3.2-1B-Instruct \
  --max-seq 256 \
  --device cpu \
  --no-run-baseline \
  --run-multiturn \
  --append-bucket 16


python -m tico.quantization.recipes.debug.static_llama_runtime \
  --model /mnt/sdb/model/Llama-3.2-1B-Instruct \
  --max-seq 256 \
  --padding-side right \
  --device cpu \
  --no-run-baseline \
  --run-multiturn \
  --multiturn-first-prompt $'User: What is the capital of France?\nAssistant:' \
  --multiturn-second-turn $'\nUser: What country is that city in?\nAssistant:' \
  --multiturn-decode-steps-before-append 4 \
  --multiturn-decode-steps-after-append 4 \
  --append-bucket 16 \
  --compare-append-prefill-with-decode-loop

```

```bash
...
====================================================================================================
Multi-turn post-append decode logits step 1
mean|diff| = 0.02939297
 max|diff| = 0.10098839
PEIR       = 0.340664 %
====================================================================================================
Multi-turn post-append decode logits step 2
mean|diff| = 0.03495336
 max|diff| = 0.17215163
PEIR       = 0.569002 %
====================================================================================================
Multi-turn post-append decode logits step 3
mean|diff| = 0.02905156
 max|diff| = 0.19826031
PEIR       = 0.583857 %
====================================================================================================
Multi-turn post-append decode logits step 4
mean|diff| = 0.01603191
 max|diff| = 0.09334183
PEIR       = 0.341133 %
====================================================================================================
Multi-turn generated text:
User: What is the capital of France?
Assistant: The capital of France
User: What country is that city in?
Assistant: France
User:

```

Related: #812
TICO-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: tico/quantization/wrapq/wrappers/llama/export_adapters.py [hunk 1]
```diff
@@ -323,6 +323,65 @@ def forward(
         return hidden, new_k, new_v
 
 
+class LlamaAttentionAppendPrefillExportAdapter(nn.Module):
```
Comment (dvsav) [inline tico/quantization/wrapq/wrappers/llama/export_adapters.py]: Couldn't find any references to this class.

File: tico/quantization/wrapq/wrappers/llama/export_adapters.py [hunk 2]
```diff
@@ -426,6 +485,69 @@ def forward(
         return hidden, new_k, new_v
 
 
+class LlamaDecoderLayerAppendPrefillExportAdapter(nn.Module):
```
Comment (dvsav) [inline tico/quantization/wrapq/wrappers/llama/export_adapters.py]: This class looks absolutely identical to `LlamaDecoderLayerDecodeExportAdapter`. Can we simply reuse the latter one?

File: tico/quantization/wrapq/wrappers/llama/export_adapters.py [hunk 3]
```diff
@@ -323,6 +323,65 @@ def forward(
...
+    --------------
+    - hidden_states:        (B, Q, D)
+    - position_embeddings:  (B, Q, head_dim)
+    - attention_mask:       (B, Q, K)
+    - past_key_value:       (B, num_kv_heads, K - Q, head_dim)
+
+    Return contract
+    ---------------
+    - return_kv=True:
+        (hidden_states, (new_key, new_value))
```
Comment (dvsav) [inline tico/quantization/wrapq/wrappers/llama/export_adapters.py]: The actual code returns a flat tuple `(hidden_states, new_key, new_value)`, not a nested tuple.

**Comment by dvsav** [comment]
### (Optional) Add unit tests

The PR adds a lot of new functionality (new export mode, new adapters, multi-turn runtime with attention mask construction, cache management, chunked append) but includes no unit tests. The only validation is the debug script `static_llama_runtime.py`. Key areas needing tests:
- `_build_append_prefill_attention_mask` correctness (causal within new block, masking of gap, padded rows).
- `append_prefill` cache write-back correctness.
- `append_prefill_chunked` with chunks smaller than bucket size.
- Export of append-prefill graphs at various bucket sizes.

File: tico/quantization/wrapq/wrappers/llama/export_adapters.py [hunk 4]
```diff
@@ -426,6 +485,69 @@ def forward(
...
+      where new_key/new_value are the delta KV tensors for the current block.
+
+    - return_kv=False:
+        hidden_states
+    """
+
+    def __init__(self, wrapped, *, return_kv: bool = True):
+        super().__init__()
+        self.wrapped = wrapped
+        self.wrapped.return_type = "tuple"
```
Comment (dvsav) [inline tico/quantization/wrapq/wrappers/llama/export_adapters.py:401]: ### (Optional) `return_type = "tuple"` mutation in `__init__` is fragile

All three decoder-layer adapters set `self.wrapped.return_type = "tuple"` in `__init__`. Since the runtime creates all three adapters from the same wrapped module, the last `__init__` wins. This works today because all set the same value, but it's a latent bug if a future adapter sets a different `return_type`. Consider setting `return_type` in `forward()` or making it a parameter of the adapter rather than mutating the wrapped module.

**Comment by dvsav** [comment]
### (Optional) Implement bucket selection logic

The [issue specification](https://github.com/Samsung/TICO/issues/812) describes runtime bucket selection: "The runtime selects the smallest usable bucket satisfying: `T <= Q` and `P + Q <= K`." The current `append_prefill_chunked` uses a **fixed** `bucket_size` for all chunks. A production runtime should select the smallest bucket that fits each chunk, and handle the case where no bucket fits (context-length policy). The current implementation (`_normalize_append_prefill_buckets`) raises `ValueError` when capacity is exceeded, which is acceptable for a first version but should be documented as a limitation.

**Review by dvsav — CHANGES_REQUESTED** [review]

File: tico/quantization/wrapq/wrappers/llama/export_adapters.py (hunk 2 above)
Comment (mhs4670go) [reply tico/quantization/wrapq/wrappers/llama/export_adapters.py]: Agreed. I'll remove the duplicate class and reuse it while keeping `append_prefill` as a separate export mode.

**Comment by mhs4670go** [comment]
@dvsav PTAL:)

**Review by dvsav — DISMISSED** [review]
👍 LGTM. Thank you!

**Review by Torrero — DISMISSED** [review]
LGTM

**Comment by mhs4670go** [comment]
@dvsav @Torrero I've just rebased the PR. PTAL

**Review by dvsav — APPROVED** [review]
👍
